app/containers/App/selectors.js

```diff
@@ -6,10 +6,9 @@ import { createSelector } from '@reduxjs/toolkit';
 
 const selectRouter = state => state.router;
 
-const makeSelectLocation = () =>
-  createSelector(
-    selectRouter,
-    routerState => routerState.location,
-  );
+const selectLocation = createSelector(
```

Comment:
hi @joeyparis - is there a reason you removed the closures?

we wrap selectors in factories so that each selector instance maintains a private cache for memoization when using across different components, see:

https://github.com/reduxjs/reselect#sharing-selectors-with-props-across-multiple-component-instances

Reply:
Hi @justingreenberg, I was really strictly following [this comment](https://github.com/react-boilerplate/react-boilerplate/issues/2806#issuecomment-554855162), which was one of the **To Do**'s. So honestly I can't say I put any deep thought into the use of factories or lack thereof. If you feel there's an alternative way to use factories, and align with the to-do/comment I'm all ears!

Reply:
Right, and per my comments, the existing code is doing that unnecessarily.  There's no reason to have multiple selector instances all reading the exact same single value.  The factory pattern is only needed when you will be calling the "same" selector with varying inputs, such as `const todo = selectTodoById(state, ownProps.id)`.  In that case, the inconsistent parameters would cause a single selector instance to not memoize at all, so you need separate instances (usually per component).

I talked about this in more detail here:

https://blog.isquaredsoftware.com/2017/12/idiomatic-redux-using-reselect-selectors/